Support for projects with `allowJs: true`

Currently when using `dts-bundle-generator` in a project with `"allowJs": true` set in their `tsconfig.json` that has JavaScript files (`.js` or `.mjs`) in the project, the following error will be generated when running `dts-bundle-generator`:

```
error TS5055: Cannot write file '<filename>' because it would overwrite input file.
```

A failing test case can be found here: https://github.com/timocov/dts-bundle-generator/compare/master...josh-:dts-bundle-generator:allowjs-test-case?expand=1.

Did you have any thoughts on supporting such project configuration? In order to resolve this error would support for `outDir` here be required? https://github.com/timocov/dts-bundle-generator/blob/a0cb4e4e8332f95ad2cce232ccaca1c07064e6f5/src/compile-dts.ts#L21

--- a/src/compile-dts.ts
+++ b/src/compile-dts.ts
@@ -96,10 +96,14 @@ function changeExtensionToDts(fileName: string): string {
 function getDeclarationFiles(rootFiles: readonly string[], compilerOptions: ts.CompilerOptions): Map<string, string> {
 	// we must pass `declaration: true` and `noEmit: false` if we want to generate declaration files
 	// see https://github.com/microsoft/TypeScript/issues/24002#issuecomment-550549393
+	// also, we don't want to generate anything apart from declarations so that's why `emitDeclarationOnly: true` is here
+	// it allows to run the tool for projects with allowJs flag enabled to avoid errors like:
+	// error TS5055: Cannot write file '<filename>' because it would overwrite input file.
 	compilerOptions = {
 		...compilerOptions,
 		noEmit: false,
 		declaration: true,
+		emitDeclarationOnly: true,
 	};
 
 	const program = ts.createProgram(rootFiles, compilerOptions);

--- a/tests/e2e/test-cases/allow-js/config.ts
+++ b/tests/e2e/test-cases/allow-js/config.ts
@@ -1,0 +1,5 @@
+import { TestCaseConfig } from '../../test-cases/test-case-config';
+
+const config: TestCaseConfig = {};
+
+export = config;

--- a/tests/e2e/test-cases/allow-js/hello.js
+++ b/tests/e2e/test-cases/allow-js/hello.js
@@ -1,0 +1,3 @@
+const hello = { test: 1 };
+
+export { hello }

--- a/tests/e2e/test-cases/allow-js/input.ts
+++ b/tests/e2e/test-cases/allow-js/input.ts
@@ -1,0 +1,5 @@
+import { hello } from "./hello.js";
+
+export function test() {
+	return hello
+}

--- a/tests/e2e/test-cases/allow-js/output.d.ts
+++ b/tests/e2e/test-cases/allow-js/output.d.ts
@@ -1,0 +1,5 @@
+export declare function test(): {
+	test: number;
+};
+
+export {};

--- a/tests/e2e/test-cases/allow-js/tsconfig.json
+++ b/tests/e2e/test-cases/allow-js/tsconfig.json
@@ -1,0 +1,6 @@
+{
+	"extends": "../tsconfig.json",
+	"compilerOptions": {
+		"allowJs": true
+	}
+}

